### ISSUE TITLE
Separate users' rights from accounts' rights

### ISSUE THREAD

**Issue by dannycolin** [AlternC/AlternC#313]
Hi folks, 

I'm opening this issue to discuss about how we could implement a better fine-grained access to accounts' functionalities. Suggestions and critcs are welcomed :).

## Actual behaviour

At the moment, alternc's accounts can only be managed by "one person" by that I mean there's only one credential to connect to an account. This behavior means that we can't delegate administrative
rights to a specifc domain.

## Expected behaviour 

It would be great to separate the logic of users' access rights and accounts. Here's a hypothetical scenario exposing that : 

User1 can manage domainA and domainB
User2 can only manage domainA

I think that could be a good first step. Of course, after that, we could add more fine-grained access like
giving users access to a specific database only or access to create emails but not add domains and etcetera.

**Comment by innoticFR** [comment]
Rights should be linked to "projects". "Projects" being a meta-entity which should

**Comment by innoticFR** [comment]
from my point of view:

1. Rights should be linked to "projects", "projects" being a meta-entity which contains multiple domains, databases, e-mail account, ...

2. Users might be able to be involved in "projects" from many base accounts (like github's accounts)

**Comment by dannycolin** [comment]
I think the organization/teams à la GitHub would be a better models than the projects one. Gandi also use something pretty similar to that [1].

[1] https://docs.gandi.net/en/sharing_products/index.html

**Comment by innoticFR** [comment]
I don't think it's incompatible ;-)

**Comment by dannycolin** [comment]
Of course, you can have both but the more things we add the more code we need to do :).
